apple/testing/default_runner/ios_xctestrun_runner.template.sh

```diff
@@ -80,11 +80,44 @@ for single_test_env in ${test_env//,/ }; do
   passthrough_env+=("SIMCTL_CHILD_$key=$value")
 done
 
+xcrun_target_app_path=""
+xcrun_is_xctrunner_hosted_bundle="false"
+xcrun_is_ui_test_bundle="false"
+test_type="%(test_type)s"
 if [[ -n "$test_host_path" ]]; then
   xctestrun_test_host_path="__TESTROOT__/$test_host_name.app"
   xctestrun_test_host_based=true
   # If this is set in the case there is no test host, some tests hang indefinitely
   xctestrun_env+="<key>XCInjectBundleInto</key><string>$(escape "__TESTHOST__/$test_host_name.app/$test_host_name")</string>"
+
+  if [[ "$test_type" = "XCUITEST" ]]; then
+    xcrun_is_xctrunner_hosted_bundle="true"
+    xcrun_is_ui_test_bundle="true"
+    xcrun_target_app_path="$xctestrun_test_host_path"
+    # If ui testing is enabled we need to copy out the XCTRunner app, update its info.plist accordingly and finally
+    # copy over the needed frameworks to enable ui testing
+    readonly runner_app_name="$test_bundle_name-Runner"
+    readonly runner_app="$runner_app_name.app"
+    readonly runner_app_destination="$test_tmp_dir/$runner_app"
+    libraries_path="$(xcode-select -p)/Platforms/iPhoneSimulator.platform/Developer/Library"
+    ditto "$libraries_path/Xcode/Agents/XCTRunner.app" "$runner_app_destination"
+    mv "$runner_app_destination/XCTRunner" "$runner_app_destination/$runner_app_name"
```

Comment:
could probably do this in 1 command instead with `cp -R` or something?

Reply:
We do need the rename I believe though? cp -R would copy the app as is.

Reply:
not if the target destination doesn't exist, `cp -R foo/bar.app baz/qux.app` should work fine as long as `qux.app` doesn't exist

Reply:
Should we then just use `XCTRunner.app/XCTRunner` as the bk arh name and update the Info.plist as such. I was trying to match xcode with the `PRODUCTNAME-Runner.app/PRODUCTNAME-Runner` naming

Reply:
i think that would be fine since our directory is unique. I think in Xcode they'd have to update it like that since derived data could contain multiple runner apps for different UI test targets